src/hooks/useDownShift.ts

```diff
@@ -24,21 +24,35 @@ const useDownShift: UseDownShift = ({ setQuery, items, onSubmit, cioClient, prev
     itemToString: (item) => item?.value || '',
     onSelectedItemChange({ selectedItem }) {
       if (selectedItem) {
-        if (selectedItem?.section === 'Search Suggestions') {
-          setQuery(selectedItem.value || '');
-        }
         if (selectedItem?.value) {
           if (onSubmit) onSubmit({ item: selectedItem, originalQuery: previousQuery });
           try {
-            if (!selectedItem?.data?.url) {
+            if (selectedItem?.section === 'Search Suggestions') {
+              setQuery(selectedItem.value || '');
               cioClient?.tracker.trackSearchSubmit(selectedItem.value, {
                 originalQuery: previousQuery,
               });
             }
-            cioClient?.tracker.trackAutocompleteSelect(selectedItem.value, {
-              originalQuery: previousQuery,
-              section: selectedItem.section,
-            });
+
+            // Autocomplete Select tracking
+            // Recommendation Select tracking
+            if (selectedItem.podId && selectedItem.data?.id) {
+              cioClient?.tracker.trackRecommendationClick({
+                itemName: selectedItem.value,
+                itemId: selectedItem.data.id,
+                variationId: selectedItem.data.variation_id,
+                podId: selectedItem.podId,
+                strategyId: selectedItem.strategy.id,
+                section: selectedItem.section,
+                resultId: selectedItem.result_id,
+              });
+              // Other Select tracking
+            } else {
+              cioClient?.tracker.trackAutocompleteSelect(selectedItem.value, {
```

Comment:
what sections do we want to send this for? 
 - search suggestions only? 
 - products and SS? 
 - content items? 
 - custom sections?
 
 I'm not sure what the answer is here, do either of y'all know, @mocca102 / @esezen ?
 
when we have clear direction on the question above, would love to refactor the logical branching in these blocks of code to make this crystal clear to anyone referencing or building on top of this code

Reply:
I believe all autocomplete sections including SS, Products, and custom sections (not to be confused with "custom" sections of this library, I'm talking about index sections)

Reply:
The previous logic did fire selects for user custom defined sections. Fixed here https://github.com/Constructor-io/constructorio-ui-autocomplete/pull/107/commits/55c168e54f3bc1568827cba615592e193f2fc3c1#diff-2df40d7bb6359f01b937f7c0c6a7cfbd72521a04f72477d270a0fb94073d430eR52